src/wp-includes/block-template-utils.php

```diff
@@ -295,11 +295,16 @@ function _get_block_template_file( $template_type, $slug ) {
  * @param string $template_type 'wp_template' or 'wp_template_part'.
  * @return array Template.
  */
-function _get_block_templates_files( $template_type ) {
+function _get_block_templates_files( $template_type, $slugs = array() ) {
```

Comment:
Quick review. Add new `$slugs` in docblock.

Reply:
Ah, thanks! There are probably much more of these changes (formatting, etc.) that I'll address when this PR is ready. It is just in the prototype stage to gauge impact, I need to look at other areas (see TODO list in the issue description).

I don't know if there's a way to mark a PR as "in progress" in core. In Gutenberg we have "draft PRs" and the "in progress" label.

Reply:
Ohh. You should mark PR as "draft" so others can understand that some work is still in progress and no one can review it. I hope that makes sense.

Reply:
Ah! I missed the option when first created the PR, but found it in [the sidebar](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request). Thank you :)

Reply:
Provided the relevant PHPDoc for this change at https://github.com/WordPress/wordpress-develop/pull/4097/commits/d532df39a4cd5de61d2483de4ffaa7189a679824